integrationTests/relayers/slowTests/framework/testSetup.go

```diff
@@ -357,19 +469,124 @@ func (setup *TestSetup) IsTransferDoneFromMultiversX(sender KeysHolder, receiver
+
+	expectedValue := setup.computeExpectedValueFromMvx(params)
+
+	if params.InitialSupplyValue != "" {
+		if initialSupply, ok := new(big.Int).SetString(params.InitialSupplyValue, 10); ok {
+			expectedValue.Add(expectedValue, initialSupply)
+		}
+	}
+
+	if bridgeNumber == secondBridge {
+		expectedValue.Sub(expectedValue, setup.computeExpectedValueToMvx(params))
+		expectedValue.Add(expectedValue, setup.getMvxTotalRefundAmountForToken(params)) // lock possible refund amount from failed SC call on Mvx
+	}
+
+	return lockedTokens.String() == expectedValue.String()
+}
+
+func (setup *TestSetup) checkMvxBurnedTokenBalance(params TestTokenParams) bool {
+	token := setup.GetTokenData(params.AbstractTokenIdentifier)
+	burnedTokens := setup.MultiversxHandler.GetBurnedAmountForToken(setup.Ctx, token.MvxChainSpecificToken)
+
+	expectedValue := setup.computeExpectedValueFromMvx(params)
+
+	if params.IsNativeOnMvX {
+		if params.InitialSupplyValue != "" {
+			if initialSupply, ok := new(big.Int).SetString(params.InitialSupplyValue, 10); ok {
+				expectedValue.Add(expectedValue, initialSupply)
+			}
+		}
+	} else {
+		expectedValue.Add(expectedValue, setup.getMvxTotalRefundAmountForToken(params)) // burn possible refund amount to be bridged back to Eth
+	}
+
+	return burnedTokens.String() == expectedValue.String()
+}
 
-	expectedEsdtSafe := big.NewInt(0).Add(initialBalanceForSafe, params.ESDTSafeExtraBalance)
-	balanceForSafe := setup.MultiversxHandler.GetESDTChainSpecificTokenBalance(setup.Ctx, setup.MultiversxHandler.SafeAddress, params.AbstractTokenIdentifier)
-	isSafeContractOnCorrectBalance := expectedEsdtSafe.String() == balanceForSafe.String()
+func (setup *TestSetup) checkEthMintedBalanceForToken(params TestTokenParams) bool {
+	token := setup.GetTokenData(params.AbstractTokenIdentifier)
+	mintedTokens := setup.EthereumHandler.GetMintBalanceForToken(setup.Ctx, token.EthErc20Address)
 
-	return okSender && okReceiver && isSafeContractOnCorrectBalance
+	expectedValue := setup.computeExpectedValueFromMvx(params)
+
+	if params.IsNativeOnMvX {
+		if params.InitialSupplyValue != "" {
+			if initialSupply, ok := new(big.Int).SetString(params.InitialSupplyValue, 10); ok {
+				expectedValue.Add(expectedValue, initialSupply)
+			}
+		}
+	} else {
+		expectedValue.Add(expectedValue, setup.getMvxTotalRefundAmountForToken(params)) // mint possible refund amount from failed SC call on Mvx
+	}
+
+	return mintedTokens.String() == expectedValue.String()
+}
+
+func (setup *TestSetup) getMvxTotalRefundAmountForToken(params TestTokenParams) *big.Int {
+	totalRefund := big.NewInt(0)
+	for _, operation := range params.TestOperations {
+		if len(operation.MvxSCCallData) > 0 || operation.MvxForceSCCall {
+			if operation.MvxFaultySCCall {
+				// the balance should be bridged back to the receiver on Ethereum - fee
+				totalRefund.Add(totalRefund, operation.ValueToTransferToMvx)
+				totalRefund.Sub(totalRefund, feeInt)
+			}
+		}
```

Comment:
```suggestion
		if !operation.MvxForceSCCall{
			continue
		}
		if len(operation.MvxSCCallData) == 0l {
			continue
		}
		if !operation.MvxFaultySCCall {
			continue
		}

		// the balance should be bridged back to the receiver on Ethereum - fee
		totalRefund.Add(totalRefund, operation.ValueToTransferToMvx)
		totalRefund.Sub(totalRefund, feeInt)
```

Reply:
I'm not sure these 2 versions behave the same. The condition from this if:
`if len(operation.MvxSCCallData) > 0 || operation.MvxForceSCCall {`
is passed if one of those two assertions is true.

On the other hand, using ifs in the suggested order behaves as if both conditions (operation.MvxForceSCCall being true and len(operation.MvxSCCallData) > 0) must be satisfied simultaneously

Working to refactor this logic